src/arkode/arkode_lsrkstep.c

```diff
@@ -1445,15 +1452,6 @@ int lsrkStep_TakeStepSSPs3(ARKodeMem ark_mem, sunrealtype* dsmPtr, int* nflagPtr
       N_VLinearSum(ONE, ark_mem->tempv1, ark_mem->h / rs, ark_mem->fn,
                    ark_mem->tempv1);
     }
-
-    /* apply user-supplied stage postprocessing function (if supplied) */
-    if (ark_mem->ProcessStage != NULL)
-    {
-      retval = ark_mem->ProcessStage(ark_mem->tcur + ((sunrealtype)j - rn - ONE) *
-                                                       rat * ark_mem->h,
-                                     ark_mem->ycur, ark_mem->user_data);
-      if (retval != 0) { return ARK_POSTPROCESS_STAGE_FAIL; }
-    }
```

Comment:
There was a comment saying this was removed because it's at the end of the step. Is loop only run once (always) to the get new solution? If so, why have the loop? If not, then `ProcessStage` needs to be called on all but the last iteration.

Reply:
Very good catch. I have finished reviewing.